java-viewer/src/main/java/org/sonar/java/viewer/DotGraph.java

```diff
@@ -88,20 +92,55 @@ public static String name(@Nullable Highlighting highlighting) {
     }
   }
 
-  public abstract static class Node extends DotDataProvider {
+  private abstract static class DotElement {
```

Comment:
Not found of this factorization by inheritance here. This seems rather artificial and seems to complicate things a bit.